saleor/graphql/order/types.py

```diff
@@ -499,8 +499,9 @@ def resolve_available_shipping_methods(root: models.Order, _info):
 
     @staticmethod
     def resolve_invoices(root: models.Order, info):
-        user = info.context.user
-        if user == root.user or user.has_perm(OrderPermissions.MANAGE_ORDERS):
+        if info.operation.name.value == "OrderByToken" or info.context.user.has_perm(
+            OrderPermissions.MANAGE_ORDERS
+        ):
```

Comment:
Maby we can use a similar approach like in `resolve_checkout`?

Have we tests for this resolver as an anonymous customer, logged-in customer, staff user, and app? 

```suggestion
        # resolve invoice for anonymous customer
        if root is None or order.user is None:
            return PermissionDenied()

        # resolve invoice for logged-in customer
        if order.user == info.context.user:
            return root.invoices.all()
            
        # resolve checkout for staff user or app
        requester = get_user_or_app_from_context(info.context)
        if requester.has_perm(OrderPermissions.MANAGE_ORDERS):
```

Reply:
Have we tests for this resolver as an anonymous customer, logged-in customer, staff user, and app?

Reply:
We shouldn't hardcode names of queries here. I think it was correct last time I check this. Something like that should work (we have the same pattern is for `resolve_user` resolver in this and few other types):
```
user = info.context.user
if user == root.user or user.has_perm(AccountPermissions.MANAGE_USERS):
    return root.invoices.all()
raise PermissionDenied()
```
This way we only return invoices if the requesting user is also the owner of the order or a staff member with MANAGE_USERS.

Reply:
We have some tests for that.

Reply:
@maarcingebala If an app tries to get an invoice resolver will raise PermissionDenied.

Reply:
I'm fine with adding support for apps here, but there are anyway more places in the entire API that don't support that; for example `resolve_user` function in this type.